### PR TITLE
ViewLocationExpanderContext should have a flag that indicates if the lookup is a partial

### LINKED ISSUE #1212: ViewLocationExpanderContext should have a flag that indicates if the lookup is for a full or partial view
Based on https://github.com/aspnet/Mvc/issues/948, Partial is another pivot on which a user could change view locations. `RazorViewEngine` uses the same locations for partials as regular views, but having this flag allows for greater customization. We would also need to account for this flag in the cache.

### PR DESCRIPTION
**Pull request by pranavkm**
lookup is for a full or partial view

Fixes #1212

### CODE REVIEW

File: src/Microsoft.AspNet.Mvc.Razor/DefaultViewLocationCache.cs [hunk 1]
```diff
@@ -50,9 +50,11 @@ internal static string GenerateKey(ViewLocationExpanderContext context)
             var routeValues = context.ActionContext.RouteData.Values;
             var controller = routeValues.GetValueOrDefault<string>(RazorViewEngine.ControllerKey);
 
-            // format is "{viewName}:{controllerName}:{areaName}:"
+            // format is "{viewName}:{isPartial}:{controllerName}:{areaName}:"
             keyBuilder.Append(context.ViewName)
                       .Append(CacheKeySeparator)
+                      .Append(context.IsPartial ? 1 : 0)
```
Comment (rynowak) [inline src/Microsoft.AspNet.Mvc.Razor/DefaultViewLocationCache.cs]: Consider if it's worth using something other than a digit here to keep this readable. I don't think it matters much but the thought occurred to me.

File: test/Microsoft.AspNet.Mvc.FunctionalTests/ViewEngineTests.cs [hunk 2]
```diff
@@ -165,6 +165,33 @@ public static IEnumerable<object[]> RazorViewEngine_UsesAllExpandedPathsToLookFo
...
+                {
+                    { "Index", "<expander-view><shared-views>/Shared-Views/ExpanderViews/_ExpanderPartial.cshtml</shared-views></expander-view>" },
+                    { "Partial", "<shared-views>/Shared-Views/ExpanderViews/_ExpanderPartial.cshtml</shared-views>" }
+                };
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(ViewLocationExpanders_PassesInIsPartialToViewLocationExpanderContextData))]
+        public async Task ViewLocationExpanders_PassesInIsPartialToViewLocationExpanderContext(string action, string expected)
```
Comment (rynowak) [inline test/Microsoft.AspNet.Mvc.FunctionalTests/ViewEngineTests.cs]: long line

**Comment by rynowak** [comment]
:shipit:

File: src/Microsoft.AspNet.Mvc.Razor/DefaultViewLocationCache.cs (hunk 1 above)
Comment (pranavkm) [inline src/Microsoft.AspNet.Mvc.Razor/DefaultViewLocationCache.cs]: Sure.

**Pull request merged**